### DISCUSSION TITLE
Attribute Types

### DISCUSSION THREAD

**Discussion by rawkode** [cuelang/cue#898] [Q&A] [answered]
I'm looking to add a `@template(1)` or `@template("David")` attribute to some Cue schemata.

The plan is to add a `blox new Definition` command to our CLI, that will consume these attributes.

A `cue.Value` is built up and rendered to JSON. However, everything within an attribute is considered a string, from what I can tell.

I've noticed that if I `fmt.Println` on the `cue.Value` returned from `fieldIterator.Value()` it prints `string` or `int`, but `.Kind()` returns bottom.

I've also tried getting the `ast.Node` from the `fieldsIterator.Value().Source()`, but it seems this returns the source for the label and the field.

I was assuming I could maybe `switch` on `.Kind()` to cast the string myself, but I've gotten a little lost now.

`.Source()` approach:

```go
		fmt.Println(fieldValue)
		node := fieldsIterator.Value().Source()

		switch v := node.(type) {
		case *ast.Ident:
			fmt.Println("IDENT", v.Name)
		default:
			fmt.Println("NODE", v) # Always get this, with label and value; only expecting value
		}
```

`.Kind()` approach always returns `_|_`
```go
		switch fieldsIterator.Value().Kind() {
		case cue.StringKind:
			fmt.Println("STRING BITCHES")
		case cue.IntKind:
			fmt.Println("INT")
		default:
			fmt.Println(fieldValue.Kind())
		}
```

Am I using these incorrectly?

**Comment by verdverm** [comment, marked as answer]
Try `IncompleteKind()` instead, this will return the underlying type for fields which are still incomplete.

**Reply by rawkode** [reply]
Thanks

**Comment by cueckoo** [comment]
This discussion has been migrated to https://github.com/cue-lang/cue/issues/898.

For more details about CUE's migration to a new home, please see https://github.com/cue-lang/cue/issues/1078.
